packages/headless/src/controllers/commerce/context/cart/headless-cart.ts

```diff
@@ -170,9 +171,9 @@ export function buildCart(engine: CommerceEngine, props: CartProps = {}): Cart {
       dispatch(setItems([]));
     },
 
-    updateItem(item: CartItem) {
+    async updateItem(item: CartItem) {
       if (isNewQuantityDifferent(item.productId, item.quantity)) {
```

Comment:
I don't have a strong opinion on that matter. If we set the logic within the `logCartAction` should we change the name?  It wouldn't be true that we log an action every time the `updateItem` function is called, as it should be called only if the quantity between the current and the previous item differs.

Reply:
Fair point on the name, in that case maybe it's ok. We could say that if someone is using the action, it is on them to use it correctly. If they send events with 0 quantity, they'll be sent but get rejected.

Perhaps it's a question of how other actions behave? Do they contain defensive checks? If not, let's keep things as-is ✔️

Reply:
That's a good point, it could be an indication to the implementer that their app has some flaws. 

@fbeaudoincoveo I think you added this check; do we usually prevent sending failing events in Headless (for example, an cartAction with a 0 quantity)?

Reply:
One example I can think of is the headless-core-generated-answer.

For instance, when you call the .like() method on this controller, it will only dispatch the "likeGeneratedAnswer" and "logLikeGeneratedAnswer" actions if "liked" isn't already true in the controller's state.

In this case, the defensive check is in the controlle, but it could arguably be done in the analytics action and in the reducer instead, though.

Now, in general I don't think we should prevent someone from emitting an invalid analytics event if they dispatch an action directly if that's what they want to do. However, if the analytics action is emitted as a side effect of calling a controller method, it may not be a bad idea to have some safeguards.

Reply:
Ok, sounds like the consistent thing is to keep things as they are:
- Controller contains defensive logic
- Dispatching an action will always send an event. There is no defensive logic, which is ok, because the backend will mark them as invalid.